### DISCUSSION TITLE
Battery discharge and disable power adapter

### DISCUSSION THREAD

**Discussion by drewluu** [mhaeuser/Battery-Toolkit#23] [Q&A]
Hi, If I disable power adapter to discharge to say zero (when calibrating my battery), will this app prevent it from charging up from zero?

**Comment by mhaeuser** [comment]
Depends on what you mean. It will deplete to 0% and then turn off. SMC configuration with the new Apple Silicon Macs is volatile, so turning off automatically clears any changes made by the tool. The tool also does not remember the power adapter disable state, so it should charge back up after you turn it back on. But frankly, I haven’t properly tested this yet.

**Comment by Shnub** [comment]
I went ahead and tested this just now, and I can confirm it behaved as @mhaeuser described it. I disabled the power adapter in BatteryToolkit, disconnected it shortly before the battery was empty (since I wanted the laptop to cool down a bit after discharging), and let the battery drain to 0 % (completely empty, not even the "Connect power" graphic showing when trying to wake). After reconnecting power, it booted up and charged normally, with the power adapter again being enabled in BatteryToolkit.

I did have the "Prevent sleep when power adapter is disabled option" enabled (as I believe is the default). Presumably, without this setting, the laptop would not discharge completely (macOS "soft" 0 % instead of actual empty battery) and instead go to sleep, so I don't know if that would suffice to re-enable the power adapter. If I understand correctly, it just re-enables due to SMC configuration being lost with (re)boots, so this might not happen when the laptop only goes to sleep with some residual charge...? But if so, you'd presumably still end up with an actual empty battery eventually because the disabled power adapter would ensure continued discharge with any attempt to wake the machine. So it might behave the same in the end.

**Reply by mhaeuser** [reply]
Thanks for testing! It actually turns off properly when running out, sleep is not entered. The charging UI and things like Find My are provided by a minimal pre-boot environment, not sleeping macOS. So this looks viable at least, thanks!

**Reply by Shnub** [reply]
Ah. But if you drain the battery normally and plug in the charger while that UI _is_ still shown, the laptop will wake from (or at least restore from?) sleep... won't it? At least I don't recall an Apple laptop doing a full boot in that case, though I encounter it so rarely that I'm really not sure.

Anyway, doesn't affect BatteryToolkit then, great.
